src/api/APIFactory.js

```diff
@@ -264,6 +276,34 @@ class APIFactory {
     }
 
     /**
+     * API for comments
+     *
+     * @param {boolean} shouldDestroy - true if the factory should destroy before returning the call
+     * @return {CommentsAPI} CommentsAPI instance
+     */
+    getCommentsAPI(shouldDestroy: boolean): CommentsAPI {
+        if (shouldDestroy) {
```

Comment:
Not sure if we have a use case yet, but I am thinking we may have to extend the destroying part to destroy itself only. In other words, getting comments should destroy only getting prior comments but not getting tasks.

This all works fine right now, since the `file` fetching API destroys everything, so getting comments for one file and comments for next file don't overlap.

Reply:
Or maybe we will never have use case for smaller APIs to destroy anything, and only the File/Folder api should destroy since they logically represent working on a brand new file or folder which will destroy all other sub apis.

Reply:
Yeah I think that makes sense. Basically just destroy the old API, and not all of them. For now, I dont think there should be any issues but should think about the scenarios of if individual API's will have to be destroyed. The only one I can think of is collaborations for now